### DISCUSSION TITLE
some needs

### DISCUSSION THREAD

**Discussion by YLQY** [snakers4/silero-vad#177] [Q&A] [answered]
If I need power consumption or memory for this VaD. Can we give the overall model code? We can try to quantify and other operations. Or if we want this VAD to adapt to multiple scenes, how can we fine tune the model

**Comment by snakers4** [comment, marked as answer]
Hi,

As for memory consumption - the best way is just to test the models.
As for tuning - the models are not locked in any way from tuning, though we do not provide tuning code.

**Reply by snakers4** [reply]
Please beware of this - https://github.com/snakers4/silero-vad/discussions/173

**Reply by YLQY** [reply]
Hi,Thank you for your answer

Is this memory leak the python version? Is it the memory leak caused by the model or the test environment? When we did the speech recognition model again, we used the libtorch Library in C + + to load the model. It is found that there will also be some memory leakage in libtorch library. The preliminary inspection is that the library of MKL is used in libtorch for matrix operation. It seems that there will be memory leakage in MKL in a specific scenario. After that, our solution is to change a reasoning platform.

**Reply by snakers4** [reply]
This is python, the leak is caused by the fact that you have to disable gradients in each process and thread for pytorch. We did not hardcode it into the model for versatility reasons, in case someone would want to tune some parts of the model.

**Reply by snakers4** [reply]
Also I believe disabling gradients is supported in JIT just recently and if we imposed some restrictions on the models, someone always found some edge case.

**Reply by YLQY** [reply]
Hi,

Has the VAD considered publishing the model in the later stage? If not, I can only reproduce it through the information printed by JIT model and Netron. I think it will be difficult. The leader of the company said: the main model is a black box for us,
  :(

**Reply by snakers4** [reply]
Can you please paraphrase? What is later stage? What company, which leader?

**Reply by YLQY** [reply]
Hi,

Later stage: in the next few months or half a year
Company: vivo
Leader: this may be inconvenient to say

**Reply by snakers4** [reply]
I am not sure if I understand you
Can you please be more specific?

**Reply by YLQY** [reply]
The simple point is: in the future, have you considered publishing the network results of this model?
:)

**Reply by snakers4** [reply]
Can you explain, what are "network results"?

**Reply by YLQY** [reply]
It is the code of the model class.
:)

For example: a python file with the following contents:
import torch
import torch.nn as nn

class vad(nn.Module):
  ......

**Reply by snakers4** [reply]
Ah, you mean the PyTorch source code of the model.
We have no plans to publish those yet.

**Reply by YLQY** [reply]
Thank you for your patience:)
